### PR TITLE
Item sets/add order props to basket

### PR DESCRIPTION
**Pull request by felixgehrmann**
### All changes meet the following requirements
- [ ] Changelog entry was added
- [x] Changes have been tested by the author
- [ ] Changes have been tested by the reviewer
- [x] Plugin can be built

@plentymarkets/ceres-io

### CODE REVIEW

File: resources/js/src/app/components/basket/AddToBasket.vue [hunk 1]
```diff
@@ -420,4 +408,36 @@ export default {
         }
     }
 }
+
+function extractPropertiesAndSurcharge(orderProperties)
+{
+    let totalSurcharge = 0;
+    const orderParams = orderProperties
+        .filter((orderProperty) => !isNullOrUndefined(orderProperty.property.value))
```
Comment (lks-mtzn) [inline resources/js/src/app/components/basket/AddToBasket.vue]: Zusammenfassen in ein forEach

**Review by lks-mtzn — CHANGES_REQUESTED** [review]

File: resources/js/src/app/components/basket/AddToBasket.vue [hunk 2]
```diff
@@ -290,50 +290,38 @@ export default {
...
-                        const variationId = this.$store.state.items[itemId]
-                            && this.$store.state.items[itemId].variation
-                            && this.$store.state.items[itemId].variation.documents[0].data.variation.id;
+                        const setComponent = this.$store.getters[`${itemId}/currentItemVariation`];
+
+                        const variationId = setComponent && setComponent.variation.id;
+
+                        // Extract order properties and total surcharge for set components
+                        const setComponentOrderParamsAndSurcharge = extractPropertiesAndSurcharge(
+                            setComponent.properties.filter(function(prop)
```
Comment (lks-mtzn) [inline resources/js/src/app/components/basket/AddToBasket.vue]:
```suggestion
                            setComponent.properties.filter(prop =>
```

File: resources/js/src/app/components/basket/AddToBasket.vue [hunk 3]
```diff
@@ -290,50 +290,38 @@ export default {
...
+                            {
+                                return prop.property.isOderProperty;
+                            })
```
Comment (lks-mtzn) [inline resources/js/src/app/components/basket/AddToBasket.vue]:
```suggestion
                            setComponent.properties.filter(prop => prop.property.isOderProperty)
```

File: resources/js/src/app/components/basket/AddToBasket.vue (hunk 1 above)
Comment (felixgehrmann) [reply resources/js/src/app/components/basket/AddToBasket.vue]: done.

**Review by lks-mtzn — APPROVED** [review]

**Pull request merged**